site-graph/http/http.go

```diff
@@ -0,0 +1,60 @@
+package http
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/adamplansky/go-bridge-mentoring/site-graph/cache"
+
+	"github.com/gorilla/mux"
+	"go.uber.org/zap"
+
+	"github.com/adamplansky/go-bridge-mentoring/site-graph/crawler"
+)
+
+type server struct {
+	router  *mux.Router
+	log     *zap.SugaredLogger
+	crawler *crawler.Crawler
+	cache   cache.Cache
+}
+
+func NewServer(log *zap.SugaredLogger, c cache.Cache) *server {
+	s := server{
+		router:  mux.NewRouter(),
+		log:     log,
+		crawler: crawler.New(log, c),
+		cache:   c,
+	}
+	s.routes()
```

Comment:
this function is mutating router - and it is not visible from func name

will be nicer if you do it like this
```
server {
...
router: s.newRouter(),
}

func (s *server) newRouter() mux.Router {
    r := mux.NewRouter()
    // set up routes
}
```